### PR TITLE
net/mlx5_vdpa: Added VIRTQ general object creation

### PR DESCRIPTION
**Pull request by shamoya**
Support for creation the VIRTQ general object via DEVX.
Also mlx5_vdpa_dma_map stores the GPA indirect KSM MKEY index
in the global priv data structure to simplify the VIRTQ creation,
instead of iterating over the mr_list.

Signed-off-by: Ido Shamay <idos@mellanox.com>

### CODE REVIEW

File: drivers/net/mlx5/mlx5_vdpa.c [hunk 1]
```diff
@@ -254,6 +257,95 @@ static int mlx5_vdpa_enable_promisc(struct vdpa_priv *priv)
...
+		}
+	}
+exit:
+	if (mem)
+		free(mem);
+	return gpa;
+}
+
+static int
+create_virtq(struct vdpa_priv *priv, int index, struct rte_vhost_vring *vq)
```
Comment (shahafsh) [inline drivers/net/mlx5/mlx5_vdpa.c]: maybe better create_split_virtq ?

File: drivers/net/mlx5/mlx5_vdpa.c [hunk 2]
```diff
@@ -54,6 +54,8 @@ struct mlx5_vdpa_caps {
 struct virtq_info {
 	uint32_t               rqn;
 	struct mlx5dv_devx_obj *rq_obj;
+	uint32_t                virtq_id;
```
Comment (shahafsh) [inline drivers/net/mlx5/mlx5_vdpa.c:62]: what would be the use of virtq_id?

File: drivers/net/mlx5/mlx5_vdpa.c [hunk 3]
```diff
@@ -254,6 +257,95 @@ static int mlx5_vdpa_enable_promisc(struct vdpa_priv *priv)
 	return ret;
 }
 
+/* TODO(idos): Move this to a shared location with ifcvf_vdpa driver */
```
Comment (shahafsh) [inline drivers/net/mlx5/mlx5_vdpa.c:349]: right.

**Review by shahafsh — CHANGES_REQUESTED** [review]
Hi ido,

nice work. very small comments.

File: drivers/net/mlx5/mlx5_vdpa.c (hunk 3 above)
Comment (shamoya) [reply drivers/net/mlx5/mlx5_vdpa.c:349]: I guess I can do it now by moving it lib_vhost somewhere.
Should I do it now?

File: drivers/net/mlx5/mlx5_vdpa.c (hunk 2 above)
Comment (shamoya) [reply drivers/net/mlx5/mlx5_vdpa.c:62]: For modifying the VIRTQ object (for example when in LM you will want to change the ctrl_mkey).
devx SW in mlx5_ib driver expects you to put the obj_id inside the command.

File: drivers/net/mlx5/mlx5_vdpa.c (hunk 1 above)
Comment (shamoya) [reply drivers/net/mlx5/mlx5_vdpa.c]: will change

**Comment by shahafsh** [comment]
No need to do it now.
We will do It when we push upstream.

**Pull request merged**